src/stirling/source_connectors/tcp_stats/tcp_stats_table.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <prometheus/counter.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "src/stirling/bpf_tools/bcc_wrapper.h"
+#include "src/stirling/core/output.h"
+#include "src/stirling/core/source_connector.h"
+#include "src/stirling/core/types.h"
+#include "src/stirling/source_connectors/tcp_stats/bcc_bpf_intf/tcp_stats.h"
+#include "src/stirling/source_connectors/tcp_stats/canonical_types.h"
+#include "src/stirling/utils/monitor.h"
+
+namespace px {
+namespace stirling {
+
+// clang-format off
+static constexpr DataElement kTCPTXElements[] = {
+      canonical_data_elements_net::kRemoteAddr,
+      {"cmd", "Process command line",
+       types::DataType::STRING, types::SemanticType::ST_NONE, types::PatternType::GENERAL},
+      {"bytes_sent", "The number of bytes sent to the remote endpoint(s).",
+         types::DataType::INT64, types::SemanticType::ST_BYTES, types::PatternType::METRIC_GAUGE},
+};
+
+static constexpr DataElement kTCPRXElements[] = {
+      canonical_data_elements_net::kRemoteAddr,
+      {"cmd", "Process command line",
+       types::DataType::STRING, types::SemanticType::ST_NONE, types::PatternType::GENERAL},
+      {"bytes_received", "The number of bytes received to the remote endpoint(s).",
+         types::DataType::INT64, types::SemanticType::ST_BYTES, types::PatternType::METRIC_GAUGE},
+};
+
+static constexpr DataElement kTCPRetransElements[] = {
+      canonical_data_elements_net::kRemoteAddr,
+      {"cmd", "Process command line",
+       types::DataType::STRING, types::SemanticType::ST_NONE, types::PatternType::GENERAL},
+      {"retrans", "The number of retransmissions to the remote endpoint(s).",
+         types::DataType::INT64, types::SemanticType::ST_NONE, types::PatternType::METRIC_GAUGE},
+};
+
```

Comment:
Is the port number included in the data here? If not, that may be interesting for at least two reasons we considered: 1. counting number of ports in use in a given time window, 2. showing a histogram of traffic per port in a given time window.

Reply:
Added port